src/IO/WriteBufferFromS3.cpp

```diff
@@ -164,34 +170,82 @@ void WriteBufferFromS3::writePart()
         LOG_WARNING(log, "Maximum part number in S3 protocol has reached (too many parts). Server may not accept this whole upload.");
     }
 
+    auto part_data = std::make_shared<Aws::StringStream>();
+    temporary_buffer.swap(part_data);
+
+    size_t part_number = part_tags.size() + 1;
```

Comment:
What is part_number

Reply:
Part number is a part index, starting with 1:

>  partNumber
>
>    Part number of part being uploaded. This is a positive integer between 1 and 10,000.
>
>    Required: Yes

https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPart.html